### PR TITLE
Validate zones order

### PR DESCRIPTION
**Pull request by timuthy**
**What this PR does / why we need it**:
This PR forces a zone order immutability for updates on the infrastructure and worker portions of a shoot. Since the AWS provider uses the zone index to create networks via Terraform as well as `MachineDeployments`, the index should be kept stable.

**Special notes for your reviewer**:
/cc @rfranzke - thanks for noticing!

**Release note**:

```improvement operator
The AWS validator now checks that zone orders in a shoot resources is retained (only additions are allowed).
```

### CODE REVIEW

**Comment by ialidzhikov** [comment]
`$index` is used also in the alicloud infra chart. We can keep in mind to add the same validation check also there once we introduce the validator for provider-alicloud.

**Comment by rfranzke** [comment]
@ismail can you add this info to https://github.com/gardener/gardener-extension-provider-alicloud/issues/15?

**Comment by timuthy** [comment]
As discussed with @rfranzke, I added the validation for workers as well.

File: pkg/apis/aws/validation/shoot_test.go [hunk 1]
```diff
@@ -165,10 +168,65 @@ var _ = Describe("Shoot validation", func() {
...
+				Expect(errorList).To(ConsistOf(
+					PointTo(MatchFields(IgnoreExtras, Fields{
+						"Type":  Equal(field.ErrorTypeInvalid),
+						"Field": Equal("workers[0].zones"),
+					})),
+				))
+			})
+		})
 	})
 })
```
Comment (rfranzke) [inline pkg/apis/aws/validation/shoot_test.go]: Can you add one more test that checks that adding a new zone while changing the existing zones is not allowed?

**Pull request merged**